internal/services/sources_service.go

```diff
@@ -68,6 +84,7 @@ func fetchARN(ctx context.Context, client sources.SourcesIntegration, sourceId s
 	if err != nil {
 		return "", fmt.Errorf("cannot list source authentication: %w", err)
 	}
+	_ = sources.HandleSourcesErrors(resp.StatusCode())
```

Comment:
Oh this does not work, the `return` statement in `HandleSourcesErrors` returns from that function, but not from the call site (here). That's why "error handling functions" are generally a bad idea because here you actually need to write:

```
err := sources.HandleSourcesErrors(resp.StatusCode())
if err != nil {
  return err
}
```

So basically the function is 7 lines long and to call it you need to use 4 lines. When you write it directly (inline) that is 6 lines, so you save 2 lines. I do not think this is worth it.